src/main/java/com/google/songgame/servlets/RoomServlet.java

```diff
@@ -1,19 +1,72 @@
 package com.google.songgame.servlets;
 
+import com.google.gson.Gson;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.stream.Collectors;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.hc.core5.http.ParseException;
+import java.lang.reflect.Type;
+import com.google.gson.reflect.TypeToken;
 
 @WebServlet("/room")
 public final class RoomServlet extends HttpServlet {
 
+  private final static Type MESSAGE_TYPE = new TypeToken<Map<String, String>>(){}.getType();
+  private Gson gson;
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void init() {
+    gson = new Gson();
+  }
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get user name
-    String name = request.getParameter("user-name");
-    response.setContentType("text/html;");
-    response.getWriter().println("Name: " + name);
+    Map<String, String> userProperties = readJSONFromRequest(request);
```

Comment:
It looks like this contains some changes from https://github.com/googleinterns/step105-2020/pull/26/files. Can you try merging master into this branch to see if that removes this diff?